### ISSUE TITLE
scheduler: scoring= is not honoured — callables and None silently become mean_squared_error, scorer sign/kwargs dropped

### ISSUE THREAD

**Issue by e-strauss** [deem-data/stratum#200]
`get_scoring_func` reimplements scorer dispatch by unwrapping the scorer's
`_score_func` instead of calling the scorer, and falls back to
`mean_squared_error` for anything it does not recognise:

https://github.com/deem-data/stratum/blob/main/stratum/runtime/_scheduler.py#L14

```python
def get_scoring_func(scoring):
    if type(scoring) == str:
        scoring = get_scorer(scoring)
    if type(scoring) == _Scorer:
        greater_is_better = scoring._sign > 0
        scoring_func = scoring._score_func      # <- sign and kwargs dropped
    else:
        greater_is_better = False
        scoring_func = mean_squared_error       # <- silent substitution
    return scoring_func, greater_is_better
```

Four ways this diverges from the same plan under plain skrub. Only a bare string
scorer comes out right:

| `scoring=` | stratum | skrub | |
|---|---|---|---|
| `'accuracy'` | 0.46744 | 0.46744 | ok |
| a callable `(estimator, X, y)` | 0.53256 | 0.46744 | scorer ignored, silently MSE |
| `make_scorer(fbeta_score, beta=2)` | `TypeError` | 0.45193 | scorer `_kwargs` dropped |
| `'neg_root_mean_squared_error'` | **+**0.36949 | **−**0.36949 | `_sign` not applied to the value |
| `None` (regressor) | 0.13653 | 0.98554 | MSE instead of the estimator's `.score()` (R²) |

The dangerous ones are silent. `0.53256` above is exactly `1 - 0.46744`: the
MSE of 0/1 predictions, reported where an accuracy was asked for. With
`make_scorer(f1_score, average='macro')` there is no `TypeError` to warn you --
`average` just reverts to `'binary'` and you get a different number. And every
`neg_*` scorer reports with the opposite sign to sklearn and skrub, which is easy
to miss because the *ranking* stays correct (`greater_is_better` handles the sort).

### Repro

```python
import numpy as np, pandas as pd, skrub, stratum
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.model_selection import StratifiedKFold

def my_accuracy(estimator, X, y_true):
    return accuracy_score(y_true, estimator.predict(X))

rng = np.random.default_rng(0)
df = pd.DataFrame({"a": rng.normal(size=300), "t": rng.integers(0, 3, size=300)})

with skrub.config_context(eager_data_ops=False):
    data = skrub.as_data_op(df)
    y = data["t"].skb.mark_as_y()
    X = data[["a"]].skb.mark_as_X(cv=StratifiedKFold(n_splits=3), split_kwargs={})
    pred = X.skb.apply(RandomForestClassifier(n_estimators=5, random_state=0), y=y)

stratum.set_config(scheduler=True)
print(pred.skb.make_grid_search(scoring="accuracy").results_["scores"].to_list())
print(pred.skb.make_grid_search(scoring=my_accuracy).results_["scores"].to_list())
# [0.3266...]  <- accuracy
# [1.3333...]  <- an MSE; the callable never ran
```

### Note on the constraint

`cross_validate` scores from `y_test` and the predicted `vals` and has no fitted
estimator to hand, which is presumably why the scorer gets unwrapped rather than
called -- a scorer's contract is `scorer(estimator, X, y)`. So a full fix means
keeping the estimator (or its predictions plus the response method) available at
scoring time.

Two things would help even before that:

- apply `_sign` and `_kwargs` when unwrapping a `_Scorer`, so string and
  `make_scorer` scorers match sklearn's value, not just its ordering;
- **raise** on a `scoring` that cannot be honoured instead of substituting
  `mean_squared_error`. A loud failure is recoverable; a plausible-looking
  wrong number is not.

Also note `type(scoring) == _Scorer` is exact-type, so `_PassthroughScorer`,
`_MultimetricScorer` and any `_Scorer` subclass take the MSE path too.

### Context

Found while collecting per-pipeline runtime statistics for 68 skrubified
MLE-STAR pipelines (follow-up to #199, thanks for the quick fix). One of them
passes a callable scorer, and its reported "accuracy" was an MSE > 1, which is
how this surfaced.

### Versions

stratum 0.0.0.dev2 @ 834dc02980379e8d6b97a04baeedc4ad3a888da6 · skrub 0.8.0 ·
scikit-learn 1.8.0 · Python 3.14.2

**Comment by e-strauss** [comment]
Closed in 0ac62e8
